lib/deterministicmnlist/SimplifiedMNListStore.js

```diff
@@ -0,0 +1,149 @@
+
+  if (!this.baseSimplifiedMNList) {
+    throw new Error('no initial diff has been provided');
+  }
+
+  // we only add to diffStore if more than one element is in diffArray
+  diffArray.forEach((diff, index) => {
+    if (index > 0) {
+      this.addDiff(diff);
+    }
+  });
+}
+
+/**
+ * Updates height and hash for latest block
+ * @param {number} height
+ * @param {string} hash
+ */
+SimplifiedMNListStore.prototype.updateTipInfo = function updateTipInfo(height, hash) {
+  this.tipHeight = height;
+  this.tipHash = hash;
+};
+
+/**
+ * Adds a new SimplifiedMNListDiff to the store.
+ * If SMLSTORE_MAX_DIFFS reached it drops the oldest
+ * @param {SimplifiedMNListDiff} diff
+ */
+SimplifiedMNListStore.prototype.addDiff = function addDiff(diff) {
+  if (!this.baseSimplifiedMNList) {
+    throw new Error('no initial diff has been provided');
+  }
+
+  if (this.diffStore.length >= this.maxDiffs - 1) {
+    this.baseSimplifiedMNList.applyDiff(this.diffStore[0].diff);
+    this.diffStore.shift();
+  }
+
+  const cbTx = new Transaction(diff.cbTx);
+  const { height } = cbTx.extraPayload;
+  this.diffStore.push({ height, diff });
+  this.currentSML.applyDiff(diff);
+  this.updateTipInfo(height, diff.blockHash);
+};
+
+/**
+ * Returns height for latest block
+ * @returns {number}
+ */
+SimplifiedMNListStore.prototype.getTipHeight = function getTipHeight() {
+  return this.tipHeight;
+};
+
+/**
+ * Returns hash for latest block
+ * @returns {string}
+ */
+SimplifiedMNListStore.prototype.getTipHash = function getTipHash() {
+  return this.tipHash;
+};
+
+/**
+ * Returns a SimplifiedMNListDiff entry by height
+ * @param {number} startHeight
+ * @param {number} endHeight
+ * @returns {SimplifiedMNListDiff[]}
+ */
+SimplifiedMNListStore.prototype.getSMLDiffbyHeightRange = function getSMLDiffbyHeightRange(startHeight, endHeight) {
+  return this.diffStore.filter(
+    diff => diff.height >= startHeight && diff.height <= endHeight,
+  ).map(
+    filteredDiffs => filteredDiffs.diff,
+  );
+};
+
+/**
+ * Returns a SimplifiedMNList by block height
+ * @param {number} height
+ * @returns {SimplifiedMNList}
+ */
+SimplifiedMNListStore.prototype.getSMLbyHeight = function getSMLbyHeight(height) {
+  const diffs = (this.baseHeight === height)
+    ? [] : this.getSMLDiffbyHeightRange(this.baseHeight, height);
+  if (diffs.length === 0 && this.baseHeight !== height) {
+    throw new Error('unable to construct SML at this height');
+  }
+  return this.applyDiffs(diffs);
+};
+
+/**
+ * Returns the SimplifiedMNList of the most recent block
+ * @returns {SimplifiedMNList}
+ */
+SimplifiedMNListStore.prototype.getCurrentSML = function getCurrentSML() {
+  return this.currentSML;
+};
+
+/**
+ * Applies an array of sequential SimplifiedMNListDiffs to the first base SimplifiedMNList
+ * @param {SimplifiedMNListDiff[]} diffs
+ * @returns {SimplifiedMNList}
+ */
+SimplifiedMNListStore.prototype.applyDiffs = function applyAllDiffs(diffs) {
```

Comment:
Name is confusing, it’s like we apply diffs to SMLStore. Let’s make it private and call `createListFromBaseAndDiffs`?

Reply:
Done